### PR TITLE
move fetcher and fetchsource into its own package

### PR DESCRIPTION
**Pull request by ddadlani**
Leaving the `fetcher` and the `resourceInstanceFetchSource` in the `resource` package causes import cycles when trying to do the refactor in #4265. Split them out into their own package to help with this.

Package name suggestions welcome. 

Signed-off-by: Divya Dadlani <ddadlani@pivotal.io>
Co-authored-by: Sameer Vohra <svohra@pivotal.io>

### CODE REVIEW

**Comment by deniseyu** [comment]
the package could be called `Artifact` maybe to avoid `fetcher.Fetch` - but that term is super overloaded as it is, and I'm not sure if it would add clarity or confusion at this point.

**Comment by ddadlani** [comment]
@deniseyu excellent point. I'm leaning toward leaving it as it is for 2 reasons:

1) we already do this in a bunch of places in our code (read: `worker.Worker`, `resource.Resource`)
2) the `fetcher`, `runtime`, `exec` and `worker` packages are in the process of being drastically refactored for #4265 so this is (hopefully) temporary

File: atc/atccmd/command.go [hunk 1]
```diff
@@ -6,6 +6,7 @@ import (
 	"database/sql"
 	"errors"
 	"fmt"
+	"github.com/concourse/concourse/atc/fetcher"
```
[the remark this answers is not in the document]
Comment (pivotal-jwinters) [reply atc/atccmd/command.go]: It shows you forced pushed, but it doesn't look like any of the imports changed?

**Comment by kcmannem** [comment]
good move

File: atc/atccmd/command.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (ddadlani) [reply atc/atccmd/command.go]: should be fixed now

**Pull request merged**